File: src/main/scala/com/calclavia/graph/api/energy/NodeElectricJunction.java
package com.calclavia.graph.api.energy;

import nova.core.block.Block;

/**
 * An electric node that acts as a junction (electrical wires) in an electric circuit.
 * Constructor requirement: Provider (An instance of {@link Block}
 * @author Calclavia
 */
public interface NodeElectricJunction extends NodeElectric {

	@Override
	default String getID() {
		return "electricJunction";
	}
}

File: src/main/scala/com/calclavia/graph/core/electric/NodeElectricJunction.scala
package com.calclavia.graph.core.electric

import com.calclavia.graph.api.NodeProvider
import com.calclavia.graph.core.electric.component.Junction

/**
 * Wires are getNodes in the grid that will not have different terminals, but instead can connect omni-directionally.
 * Wires will be treated as junctions and collapsed.
 * @author Calclavia
 */
class NodeElectricJunction(parent: NodeProvider) extends NodeAbstractElectric(parent) with com.calclavia.graph.api.energy.NodeElectricJunction {

	var junction: Junction = null

	override def current: Double = voltage * voltage / resistance

	override def voltage: Double = junction.voltage

	override def toString: String = {
		"ElectricJunction [" + connections.size() + ", " + BigDecimal(voltage).setScale(2, BigDecimal.RoundingMode.HALF_UP) + "V]"
	}
}
